src/methods/line_search_helpers/nls_cache_util.jl

```diff
@@ -0,0 +1,85 @@
+# Date: 01/27/2025
+# Author: Christian Varner
+# Purpose: Helper functions to maintain the
+# maximum value for non-monotone line search
+
```

Comment:
Why not use the [CircularArrays](https://www.juliapackages.com/p/circulararrays) package?